controller/konnect/ops/ops.go

```diff
@@ -215,6 +217,19 @@ func Create[
 			time.Since(start),
 		)
 	}
+
+	// For ControlPlane entities, we need to check if the source type is mirror
+	// and set the mirrored condition accordingly.
+	if cp, ok := any(e).(*konnectv1alpha1.KonnectGatewayControlPlane); ok {
+		if *cp.Spec.Source == commonv1alpha1.EntitySourceMirror {
```

Comment:
The `spec.Source` is optional (and legacy cps does not have such a field), I think it is clearer and error prone to define some utility functions to get or judge the entity source.

Reply:
The `spec.source` is optional but defaulted. All the CPs will have this field set after CRDs upgrade